### PR TITLE
Champva 94284 pega reporting api connection

### PR DESCRIPTION
**Pull request by stevelong00**
## Summary

- This work is behind a feature toggle (flipper): NO
- Created a client for the Pega API with some tests to make use of it.
- We will need to retrieve data from this API in upcoming tickets.  With this ticket we have implemented a simple API client and some test code as a proof of concept.
- I work for the IVC Forms team and we own this module.

## Related issue(s)

- [The associated ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/94284)

## Testing done

- [x] New code is covered by unit tests
- Prior to this change there was no client for the Pega API.  With this change, there is a client that is only used by unit test code.
- In addition to running the test suite, verification can be performed by:
  - Enable the 'hit the production endpoint' test
  - Configure the API key in your local environment as described in the comments
  - Run this test locally
  - Verify the response using a debugger

## Screenshots
None

## What areas of the site does it impact?
None, this code is only used by unit tests so far.

## Acceptance criteria

- [x]  I added unit tests ~~and integration tests~~ for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback
None

### CODE REVIEW

**Comment by michaelclement** [comment]
@stevelong00 after troubleshooting with Cindy we found that the headers were automatically getting converted to have an uppercase first letter. Cindy updated her python lambda to grab the headers using appropriate casing and I was able to use your existing code to pull a report in the Argo staging console:

Screenshot 2024-12-23 at 14 28 01

**Review by michaelclement — APPROVED** [review]

File: config/settings.yml [hunk 1]
```diff
@@ -400,7 +400,8 @@ intent_to_file:
   prefill: true
 
 ivc_champva:
-  prefill: true
+  pega_api:
+    api_key: fake_api_key
```
Comment (michaelclement) [inline config/settings.yml:404]: I think we want that prefill line in here (for when users have save-in-progress'd data).

Comment (michaelclement) [reply config/settings.yml:404]: Also, I've added the production API key to the AWS param store under this path.

Comment (stevelong00) [reply config/settings.yml]: Thanks!  I wasn't sure if that prefill line was a placeholder or if it was doing something.

File: modules/ivc_champva/lib/pega_api/configuration.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+require 'common/client/configuration/rest'
+require 'common/client/middleware/response/raise_custom_error'
+
+module IvcChampva
+  module PegaApi
+    class Configuration < Common::Client::Configuration::REST
+      def base_path
+        'https://bt41mfpkj5.execute-api.us-gov-west-1.amazonaws.com/prod/'
+      end
```
Comment (michaelclement) [inline modules/ivc_champva/lib/pega_api/configuration.rb:11]: Not a blocker, but we may want to move this into an AWS parameter at some point in the future.

**Review by michaelclement — APPROVED** [review]
Looks great!

File: modules/ivc_champva/lib/pega_api/configuration.rb (hunk 2 above)
Comment (stevelong00) [reply modules/ivc_champva/lib/pega_api/configuration.rb:11]: I was thinking about that, but this is how paths are being stored elsewhere in [similar code](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/dhp_connected_devices/lib/fitbit/configuration.rb) so I decided to err on the side of consistency.

**Review by ryan-mcneil — APPROVED** [review]

**Pull request merged**